powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/EciesEncryptionBehavior.java

```diff
@@ -166,28 +168,44 @@ private GetEciesDecryptorResponse getEciesDecryptorParametersForApplication(GetE
                         request.getProtocolVersion());
             }
 
-            // Get master private key
-            final ApplicationEntity application = applicationVersion.getApplication();
-            final String applicationId = application.getId();
-            final MasterKeyPairEntity masterKeyPairEntity = repositoryCatalogue.getMasterKeyPairRepository().findFirstByApplicationIdOrderByTimestampCreatedDesc(applicationId);
-            if (masterKeyPairEntity == null) {
-                logger.error("Missing key pair for application ID: {}", applicationId);
-                // Rollback is not required, database is not used for writing
-                throw localizationProvider.buildExceptionForCode(ServiceError.NO_MASTER_SERVER_KEYPAIR);
-            }
+            final String temporaryKeyId = request.getTemporaryKeyId();
+            final PrivateKey privateKey;
+            if (temporaryKeyId == null) {
```

Comment:
The could would be much more readable with inverted logic `if (temporaryKeyId != null)`.

Reply:
dtto